pennylane/grouping/utils.py

```diff
@@ -0,0 +1,589 @@
+            observable.
+
+    Returns:
+        bool: true if the input observable is a Pauli word, false otherwise.
+
+    Raises:
+        TypeError: if input observable is not an Observable instance.
+    """
+
+    if not isinstance(observable, Observable):
+        raise TypeError(
+            "Expected {} instance, instead got {} instance.".format(Observable, type(observable))
+        )
+
+    pauli_word_names = ["Identity", "PauliX", "PauliY", "PauliZ"]
+    if isinstance(observable, Tensor):
+        return set(observable.name).issubset(pauli_word_names)
+
+    return observable.name in pauli_word_names
+
+
+def are_identical_pauli_words(pauli_1, pauli_2):
+    """Performs a check if two Pauli words have the same `wires` and `name` attributes.
+
+    This is a convenience function that checks if two given Tensor instances specify the same
+    Pauli word. This function only checks if both Tensor instances have the same wires and name
+    attributes, and hence won't perform any simplification to identify if the two Pauli words are
+    algebraically equivalent. For instance, this function will not identify that
+    PauliX(0) @ PauliX(0) = Identity(0), or PauliX(0) @ Identity(1) = PauliX(0), or
+    Identity(0) = Identity(1), etc.
+
+    **Usage example:**
+
+    >>> are_identical_pauli_words(qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliZ(1))
+    True
+    >>> are_identical_pauli_words(qml.PauliZ(0) @ qml.PauliZ(1), qml.PauliZ(0) @ qml.PauliX(3))
+    False
+
+    Args:
+        pauli_1 (Union[Identity, PauliX, PauliY, PauliZ, Tensor]): the first Pauli word.
+        pauli_2 (Union[Identity, PauliX, PauliY, PauliZ, Tensor]): the second Pauli word.
+
+    Returns:
+        Boolean: whether `pauli_1` and `pauli_2` have the same wires and name attributes.
+
+    Raises:
+        TypeError: if pauli_1 or pauli_2 are not Identity, PauliX, PauliY, PauliZ, or Tensor
+            instances.
+
+    """
+
+    if not (is_pauli_word(pauli_1) and is_pauli_word(pauli_2)):
+        raise TypeError(
+            "Expected Pauli word observables, instead got {} and {}.".format(pauli_1, pauli_2)
+        )
+
+    paulis_with_identity = (PauliX, PauliY, PauliZ, Identity)
+    if isinstance(pauli_1, paulis_with_identity) and isinstance(pauli_2, paulis_with_identity):
+        return (pauli_1.wires, pauli_1.name) == (pauli_2.wires, pauli_2.name)
+
+    if isinstance(pauli_1, Tensor) and isinstance(pauli_2, Tensor):
+        return set(zip(pauli_1.wires, pauli_1.name)) == set(zip(pauli_2.wires, pauli_2.name))
+
+    if isinstance(pauli_1, paulis_with_identity) and isinstance(pauli_2, Tensor):
+        return {(pauli_1.wires, pauli_1.name)} == set(zip(pauli_2.wires, pauli_2.name))
+
+    if isinstance(pauli_1, Tensor) and isinstance(pauli_2, paulis_with_identity):
+        return set(zip(pauli_1.wires, pauli_1.name)) == {(pauli_2.wires, pauli_2.name)}
+
+    return False
+
+
+def binary_symplectic_inner_prod(pauli_vec_1, pauli_vec_2):
```

Comment:
Is this function being used?

Reply:
It's not being used at the moment. It will be used quite a bit once the utility functions for the fully commuting approach are implemented. I can remove it from this PR and include it with the fully commuting approach PR once it's ready.